### DISCUSSION TITLE
How to get original Pytorch Weights and Source code of the models❓ Questions / Help / Support

### DISCUSSION THREAD

**Discussion by elcronos** [snakers4/silero-models#76] [Q&A] [answered]
Hi,

I'm currently researching on robustness of AI systems in sound classification tasks. In order to investigate how the activations some decision are made, I need to be able to modify and access some layer in the model. However, currently the models I get from Pytorch Hub are loaded as ScriptModules. I was wondering if I could get the original code of the network and their corresponding weights as a .pt file. This will help me to carry on with the experiements I need. Of course, I'm happy to cite your work.

**Comment by snakers4** [comment, marked as answer]
Hi @elcronos,

Thank you for your interest

We do not publish the original code intentionally

But for your purposes please check out the `skip` model - it is what you are looking for

**Reply by snakers4** [reply]
https://github.com/snakers4/silero-models/blob/fa75d0f7c0de6de02da62275b1817558f0e275d4/models.yml#L12

**Reply by elcronos** [reply]
Hi @snakers4,

I understand about the availability of the code. In reality I would like to be able for example to replace some layers, to add hooks to check the activations, etc. I'm not familiar with jit so I'm not sure what you mean with jit_skip model. Could you please elaborate more on that?

**Reply by snakers4** [reply]
I added skip connections to the model in the most sensible places, i.e. it outputs several feature maps

Just try and you will see

**Reply by snakers4** [reply]
Most likely you need something like this https://github.com/snakers4/silero-models/discussions/53
